src/libsystemd/sd-device/device-monitor.c

```diff
@@ -246,6 +246,8 @@ static int device_monitor_event_handler(sd_event_source *s, int fd, uint32_t rev
         if (device_monitor_receive_device(m, &device) <= 0)
                 return 0;
 
+        LOG_CONTEXT_CONSUME_STRV(device_make_log_fields(device));
```

Comment:
hmm, i am not sure i like this use here...

I mean, think about the callback function called below might start doing pretty much anything, for example a screen redraw in a GUI app. But if some graphics function then logs, should that really carry device info? I am not sure...

Using this funcitonality is good, but when doing a generic callback dispatch then that's going to far I think, because you have no clue what the callback will do and if it will issue more than just stuff directly related to the device even we have here...

Reply:
Maybe this is excessive information to attach by default. But it's still immensely useful information to know that the graphics function logged a message during the processing of a device monitor callback for device X.

You can go pretty wild with the context concept. If you want to be able to filter all log messages filtered during the execution of a particular function, just push some unique field on the context and you can filter by that in journalctl.

Reply:
Still I do not like this. And I agree with @poettering's concern in the above. Relevant discussion is here https://github.com/systemd/systemd/pull/23309#discussion_r947240763 (sorry for late response, though).